presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkLauncherCommand.java

```diff
@@ -44,8 +44,11 @@
 
     public void run()
     {
-        SparkConf sparkConfiguration = new SparkConf()
-                .setAppName("Presto");
+        SparkConf sparkConfiguration = new SparkConf();
+        if (sparkConfiguration.getOption("spark.app.name").isEmpty()) {
+            sparkConfiguration.setAppName("Presto");
+        }
```

Comment:
> How is this different from existing code? As you are creating a new SparkConf object, `spark.app.name` is going to be empty anyway. So, you will end up setting appname as "Presto". Am I missing something here?

If user set `--conf spark.app.name=APP_NAME` in `spark-submit,` `sparkConfiguration.getOption("spark.app.name").isEmpty()` will be false.

Reply:
For example:

Reply:
That is true. But, if you see in code, you are creating a new SparkConf object and trying to fetch `spark.app.name` from this object, where it will have the default value or might be null itself.  Values passed via `--conf spark.app.name` in spark-submit command will be not visible here, at least in the current code shape. 

If you want to set the appName, I would suggest adding a new option in PrestoSparkClientOptions and pass appName via that option. Then you can access it here and set it in the SparkConf object.

Reply:
Thinks for your review!